be/src/util/file_utils.cpp

```diff
@@ -33,36 +33,64 @@
 
 #include <openssl/md5.h>
 
+#include "gutil/strings/substitute.h"
+
 #include "env/env.h"
 #include "olap/file_helper.h"
 #include "util/defer_op.h"
 
 namespace doris {
 
-Status FileUtils::create_dir(const std::string& dir_path) {
-    try {
-        if (boost::filesystem::exists(dir_path.c_str())) {
-            // No need to create one
-            if (!boost::filesystem::is_directory(dir_path.c_str())) {
-                std::stringstream ss;
-                ss << "Path(" << dir_path << ") already exists, but not a directory.";
-                return Status::InternalError(ss.str());
+using strings::Substitute;
+
+Status FileUtils::create_dir(const std::string& path, Env* env) {
+    if (path.empty()) {
+        return Status::InvalidArgument(Substitute("Unknown primitive type($0)", path));
+    }
+
+    boost::filesystem::path p(path);
+
+    string partial_path;
+    for (boost::filesystem::path::iterator it = p.begin(); it != p.end(); ++it) {
+        partial_path = partial_path.empty() ? it->string() : partial_path + "/" + it->string();
+        bool is_dir = false;
+        
+        Status s = env->is_directory(partial_path, &is_dir);
```

Comment:
```suggestion
        RETURN_IF_ERROR(env->is_directory(partial_path, &is_dir));
```
better to return instantly when meeting error.

Reply:
We can't return here because the path maybe isn't exists.